src/main/kotlin/build/buf/intellij/index/BufModuleConfigIndex.kt

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buf.intellij.index
+
+import build.buf.intellij.config.BufConfig
+import build.buf.intellij.module.ModuleFullName
+import com.intellij.util.indexing.DataIndexer
+import com.intellij.util.indexing.FileBasedIndex
+import com.intellij.util.indexing.FileContent
+import com.intellij.util.indexing.ID
+import com.intellij.util.indexing.ScalarIndexExtension
+import com.intellij.util.io.IOUtil
+import com.intellij.util.io.KeyDescriptor
+import org.jetbrains.yaml.YAMLUtil
+import org.jetbrains.yaml.psi.YAMLFile
+import org.jetbrains.yaml.psi.YAMLScalar
+import org.jetbrains.yaml.psi.YAMLSequence
+import java.io.DataInput
+import java.io.DataOutput
+import java.io.IOException
+
+/**
+ * [BufModuleConfigIndex] indexes [BufModuleConfig] instances loaded from `buf.yaml` files.
+ */
+class BufModuleConfigIndex : ScalarIndexExtension<BufModuleConfig>() {
+
+    override fun getName(): ID<BufModuleConfig, Void> = BufIndexes.BUF_MODULE_CONFIG_INDEX_ID
+
+    override fun dependsOnFileContent(): Boolean = true
+
+    override fun getVersion(): Int = 1
```

Comment:
I guess the identity of the index is the fully-qualified class name? Is that why you can start this at one?

Reply:
It appears to be the combination of `getName()` and `getVersion()`.